micromanager_gui/_core_widgets/_stage_widget.py

```diff
@@ -0,0 +1,225 @@
+from itertools import chain, product, repeat
+
+from fonticon_mdi6 import MDI6
+from pymmcore_plus import DeviceType
+from qtpy.QtCore import Qt, QTimer
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QGridLayout,
+    QHBoxLayout,
+    QLabel,
+    QPushButton,
+    QSpinBox,
+    QVBoxLayout,
+    QWidget,
+)
+from superqt.fonticon import setTextIcon
+
+from micromanager_gui import _core
+
+AlignCenter = Qt.AlignmentFlag.AlignCenter
+PREFIX = MDI6.__name__.lower()
+STAGE_DEVICES = {DeviceType.Stage, DeviceType.XYStage, DeviceType.AutoFocus}
+STYLE = """
+QPushButton {
+    border: none;
+    background: transparent;
+    color: rgb(0, 180, 0);
+    font-size: 40px;
+}
+QPushButton:hover:!pressed {
+    color: rgb(0, 255, 0);
+}
+QPushButton:pressed {
+    color: rgb(0, 100, 0);
+}
+QSpinBox {
+    min-width: 35px;
+    height: 22px;
+}
+QLabel {
+    color: #999;
+}
+QCheckBox {
+    color: #999;
+}
+QCheckBox::indicator {
+    width: 11px;
+    height: 11px;
+}
+"""
+
+
+class StageWidget(QWidget):
+    # fmt: off
+    BTNS = {
+        # btn glyph                (r, c, xmag, ymag)
+        MDI6.chevron_triple_up:    (0, 3,  0,  3),
+        MDI6.chevron_double_up:    (1, 3,  0,  2),
+        MDI6.chevron_up:           (2, 3,  0,  1),
+        MDI6.chevron_down:         (4, 3,  0, -1),
+        MDI6.chevron_double_down:  (5, 3,  0, -2),
+        MDI6.chevron_triple_down:  (6, 3,  0, -3),
+        MDI6.chevron_triple_left:  (3, 0, -3,  0),
+        MDI6.chevron_double_left:  (3, 1, -2,  0),
+        MDI6.chevron_left:         (3, 2, -1,  0),
+        MDI6.chevron_right:        (3, 4,  1,  0),
+        MDI6.chevron_double_right: (3, 5,  2,  0),
+        MDI6.chevron_triple_right: (3, 6,  3,  0),
+    }
+    BTN_SIZE = 36
+    # fmt: on
+
+    def __init__(self, levels=2, device=None, mmcore=None):
+        super().__init__()
+        self._mmc = mmcore or _core.get_core_singleton()
+        self._device = device or self._mmc.getXYStageDevice()
+        self._dtype = self._mmc.getDeviceType(self._device)
+        assert self._dtype in STAGE_DEVICES, f"{self._dtype} not in {STAGE_DEVICES}"
+        self.setStyleSheet(STYLE)
+        self._snap_on_click = True  # TODO: make widget
+
+        self._step = QSpinBox()
```

Comment:
```suggestion
        self._step = QDoubleSpinBox()
```

currently is limited to 1 micron steps